awswrangler/s3/_read_parquet.py

```diff
@@ -236,457 +216,269 @@ def _read_parquet_metadata(
-        boto3_session=boto3_session,
-    ) as f:
-        pq_file: Optional[pyarrow.parquet.ParquetFile] = _pyarrow_parquet_file_wrapper(
-            source=f, read_dictionary=categories
-        )
-        if pq_file is None:
-            return 0
-        n: int = cast(int, pq_file.num_row_groups)
-        _logger.debug("Row groups count: %d", n)
-        return n
-
-
 def _read_parquet(
-    path: str,
-    version_id: Optional[str],
-    columns: Optional[List[str]],
-    categories: Optional[List[str]],
-    safe: bool,
-    map_types: bool,
-    boto3_session: Union[boto3.Session, _utils.Boto3PrimitivesType],
-    dataset: bool,
-    validate_schema: Optional[bool],
+    paths: List[str],
     path_root: Optional[str],
-    s3_additional_kwargs: Optional[Dict[str, str]],
+    schema: pa.schema,
+    columns: Optional[List[str]],
+    coerce_int96_timestamp_unit: Optional[str],
+    chunked: Union[int, bool],
+    boto3_session: Optional[boto3.Session],
     use_threads: Union[bool, int],
-    pyarrow_additional_kwargs: Optional[Dict[str, Any]] = None,
-) -> pd.DataFrame:
-    pyarrow_args = _set_default_pyarrow_additional_kwargs(pyarrow_additional_kwargs)
-    boto3_session = _utils.ensure_session(boto3_session)
-    df: pd.DataFrame = _arrowtable2df(
-        table=_read_parquet_file(
-            path=path,
+    parallelism: int,
+    version_ids: Optional[Dict[str, str]],
+    s3_additional_kwargs: Optional[Dict[str, Any]],
+    arrow_kwargs: Dict[str, Any],
+) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+    if config.distributed:
+        dataset = read_datasource(
+            datasource=ParquetDatasource(),
+            parallelism=parallelism,
+            use_threads=use_threads,
+            paths=paths,
+            schema=schema,
             columns=columns,
-            categories=categories,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            path_root=path_root,
+        )
+        return _to_modin(dataset=dataset, kwargs=arrow_kwargs)
+
+    if chunked:
+        return _read_parquet_chunked(
             boto3_session=boto3_session,
-            s3_additional_kwargs=s3_additional_kwargs,
+            paths=paths,
+            path_root=path_root,
+            columns=columns,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            chunked=chunked,
             use_threads=use_threads,
-            version_id=version_id,
-            pyarrow_additional_kwargs=pyarrow_args,
-        ),
-        categories=categories,
-        safe=safe,
-        map_types=map_types,
-        use_threads=use_threads,
-        dataset=dataset,
-        path=path,
-        path_root=path_root,
-        timestamp_as_object=pyarrow_args["timestamp_as_object"],
+            s3_additional_kwargs=s3_additional_kwargs,
+            arrow_kwargs=arrow_kwargs,
+            version_ids=version_ids,
+        )
+
+    executor = _get_executor(use_threads=use_threads)
```

Comment:
So do we not ever use executor if `config.distributed == False` or am I reading this method wrong?

Reply:
I think I may have cleared up my own confusion after reading through `datasource.parquet_datasource.py` :)